lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -268,41 +289,43 @@ BOOST_AUTO_TEST_CASE(testPiecewiseConstant) {
         };
     };
 
-    TDoubleVecVec modelBias;
-    TDoubleVecVec modelRSquared;
-
     test::CRandomNumbers rng;
     double noiseVariance{0.2};
     std::size_t trainRows{1000};
     std::size_t testRows{200};
-    std::size_t cols{6};
+    std::size_t cols{3};
     std::size_t capacity{250};
+    for (auto lossFunctionType : {TLossFunctionType::E_MseRegression /* , TLossFunctionType::E_MsleRegression,
+          TLossFunctionType::E_HuberRegression */}) {
+        TDoubleVecVec modelBias;
+        TDoubleVecVec modelRSquared;
+        std::tie(modelBias, modelRSquared) = predictAndComputeEvaluationMetrics(
```

Comment:
You're going to need some trick to ensure the target is positive for MSLE.

Reply:
Good point, I adjusted the target generators to add a large positive constant for MSLE